components/project-loader/project-loader.jsx

```diff
@@ -1,51 +1,53 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 import Service from '../../js/service.js';
 import ProjectList from '../project-list/project-list.jsx';
 import pageSettings from '../../js/app-page-settings';
 import env from "../../config/env.generated.json";
 
 const PROJECT_BATCH_SIZE = env.PROJECT_BATCH_SIZE;
 
-export default React.createClass({
-  getInitialState() {
-    return {
+class ProjectLoader extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
```

Comment:
keeping this the result of a function makes it possible to reset a component at any time, so it's worth keeping:
```
constructor(props) {
  super(props);
  this.state = this.getInitialState();
}
getInitialState() {
  return { 
    ...
  };
}
```
Keeping things separated this way allows for component resetting by calling `this.setState(this.getInitialState())`, inspecting the "what state should be at initi", etc.